types/test.ts

```diff
@@ -74,62 +74,62 @@ function testSelectorAsCombiner() {
     subSelector,
   );
 
-  // typings:expect-error
+  // $ExpectError
   selector({foo: ''});
 
-  // typings:expect-error
+  // $ExpectError
   const n: number = selector({bar: {foo: ''}});
 
   const s: string = selector({bar: {foo: ''}});
 }
 
 type Component<P> = (props: P) => any;
 
-declare function connect<S, P, R>(selector: ParametricSelector<S, P, R>):
-  (component: Component<P & R>) => Component<P>;
+declare function connect<S, R, P>(selector: Selector<S, R, P[]>):
+  (component: Component<R & S & ([P] extends [never] ? {} : P)>) => Component<P>;
```

Comment:
The `Selector` types were modified, so we needed to update this helper

Reply:
I noticed that `R` and `P` are swapped here. Would that impact back compatability?

Reply:
Not here in the test, but it might cause issues in client usage of `Selector`. This was something that I should revert though.

I changed it without realizing because I rewrote the types from scratch. I'm definitely going to fix the order here, since it's a pointless change that would just inconvenience people.